### PR TITLE
Bug fix `live_retrain_hours`

### PR DESCRIPTION
**Pull request by robcaulk**
The introduction of `buffer_train_data_candles` truncates the training data timerange. This `timerange.stopts` is used after training to define the training timestamp, which is compared later against the `live_retrain_hours` to decide if a new model should be trained or not. 

If the user buffered the training data, then this `timerange.stopts` was artificially pushed backwards, causing `live_retrain_hours` to improperly think that a new train should be started, even though a model was recently trained within the `live_retrain_hours` period. 

This PR fixes the problem.

Thanks @smarmau and @davedavesen for reporting and investigating.

### CODE REVIEW

**Review by xmatthias — APPROVED** [review]
LGTM 👍

**Comment by robcaulk** [comment]
Waiting on @smarmau confirmation before merge on this one.

**Comment by smarmau** [comment]
> Waiting on @smarmau confirmation before merge on this one.

Confirmed fix is good. Thanks @robcaulk and @xmatthias

**Pull request merged**
